bittensor/_dataset/dataset_impl.py

```diff
@@ -1,536 +1,490 @@
+        self.all_text_file_metas = all_text_file_metas
+
+    construct_text_corpus = build_datasets
+    async def async_save_json(self, 
+                              path:str,
+                              obj:Union[dict, list],
+                              include_root:bool=True) -> str:
+        """ 
+        Async save of json for storing text hashes
 
-    def get_dataset(self , file_meta):
-        r""" Either load a dataset, which is a list of hashes, from disk or download it from IPFS
         Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
-
-        Return:
-            hashes (list):
-                The hashes from the dataset downloaded from disk or IPFS.     
+            path (List[str]):
+                Axon to serve.
+            obj (bool):
+                The object to save locally
+            include_root (bool):
+                Include self.data_dir as the prefix.
+                    - if True, ths meants shortens the batch and 
+                    specializes it to be with respect to the dataset's 
+                    root path which is in ./bittensor/dataset
+            
+        Returns: 
+            path (str)
+                Path of the saved JSON
         """
-        # --- Load text from path
-        logger.success( f"Getting dataset: {file_meta['Name']}" )
         
-        hashes = self.load_hash(file_meta)
+        if include_root:
+            path = os.path.join(self.data_dir, path)
 
-        if hashes != None:
-            hashes = json.loads(hashes)
+        dir_path = os.path.dirname(path)
 
-        # --- If couldnt load from path, download text.
-        else:
-            response = self.get_ipfs_directory(self.dataset_dir, file_meta)
-            if (response != None) and (response.status_code == 200):
-                self.IPFS_fails = 0
-                hashes = response.json()
-
-                # --- Save text if the save_dataset flag is on.
-                if self.save_dataset :
-                    self.save_hash(file_meta, json.dumps(response.json()) )
-                    
-            else:
-                self.IPFS_fails += 1
-                logger.warning("Failed to get dataset".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-                return None
+        # ensure the json is the prefix
+        if path[-len('.json'):] != '.json':
+            path += '.json'
 
-        if hashes == None:
-            return None
-        else:
-            for h in hashes:
-                h['Folder'] = file_meta['Name']
-            return hashes 
+        # ensure the directory exists, make otherwise
+        if not os.path.isdir(dir_path):
+            os.makedirs(dir_path)
 
-    def get_hashes_from_dataset(self):
-        r""" Getting directories .
-        Where a directory could be leading to a data file or a directory file.
+        assert os.access( path , os.R_OK )
```

Comment:
Maybe a warning is more appropriate here, rather than halting the entire program if we can't save the data

Reply:
This should check access on `dir_path`, and we're probably more interested in `os.W_OK` rather than if it's readable. Because `path` is the path to the file (and because it doesn't exist yet), it's going to raise an error here the first time, every time.

Reply:
I left the assert but changed to W_OK.